### PR TITLE
Improve influxdb provider

### PR DESCRIPTION
**Pull request by nicolai86**
- reduce public funcs. We should not make things public that don't need to be public
- improve tests by verifying remote state
- add `influxdb_user` resource

    allows you to manage influxdb users:

    ```
    resource "influxdb_user" "admin" {
      name = "administrator"
      password = "super-secret"
      admin = true
    }
    ```

    and also database specific grants:

    ```
    resource "influxdb_user" "ro" {
      name = "read-only"
      password = "read-only"

      grant {
        database = "a"
        privilege = "read"
      }
    }
    ```

- add `influxdb_continuous_query` resource

    allows you to manage influxdb continuous queries:

    ```
    resource "influxdb_continuous_query" "minnie" {
        name = "minnie"
        database = "metrics"
        query = "SELECT min(mouse) INTO min_mouse FROM zoo GROUP BY time(30m)"
    }

    ```

this PR includes tests & documentation changes.

looking forward to your feedback!

### CODE REVIEW

File: builtin/providers/influxdb/resource_user.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+			},
+			"password": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"admin": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
```
Comment (stack72) [inline builtin/providers/influxdb/resource_user.go]: There doesn't seem to be support for an admin update. What happens if a user is an admin and then no longer is an admin?

Comment (nicolai86) [inline builtin/providers/influxdb/resource_user.go]: That should issue a `REVOKE ALL PRIVILEGES`. Will be added today,thanks

Comment (nicolai86) [inline builtin/providers/influxdb/resource_user.go]: Likewise, becoming an admin should be `GRANT ALL PRIVILEGES`

**Comment by nicolai86** [comment]
@stack72 I've pushed a version which properly handles changes in the admin flag.

**Comment by nicolai86** [comment]
I've also included the `influxdb_continuous_query` resource to make this a more feature complete PR.

**Comment by stack72** [comment]
Hi @nicolai86 

These changes look good and the tests work as expected :)

```
% make testacc TEST=./builtin/providers/influxdb TESTARGS='-run=TestAccInfluxDB'
==> Checking that code complies with gofmt requirements...
go generate $(go list ./... | grep -v /terraform/vendor/)
TF_ACC=1 go test ./builtin/providers/influxdb -v -run=TestAccInfluxDB -timeout 120m
=== RUN   TestAccInfluxDBContiuousQuery
--- PASS: TestAccInfluxDBContiuousQuery (0.07s)
=== RUN   TestAccInfluxDBDatabase
--- PASS: TestAccInfluxDBDatabase (0.05s)
=== RUN   TestAccInfluxDBUser_admin
--- PASS: TestAccInfluxDBUser_admin (0.21s)
=== RUN   TestAccInfluxDBUser_grant
--- PASS: TestAccInfluxDBUser_grant (0.26s)
=== RUN   TestAccInfluxDBUser_revoke
--- PASS: TestAccInfluxDBUser_revoke (0.22s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/influxdb	0.816s
```

**Pull request merged**